platform/ui/src/contextProviders/ContextMenuProvider.js

```diff
@@ -0,0 +1,136 @@
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useCallback,
+} from 'react';
+import PropTypes from 'prop-types';
+
+import ToolContextMenu from '../../../viewer/src/connectedComponents/ToolContextMenu';
+
+import { useDialog } from './DialogProvider';
+import { useLabellingFlow } from './LabellingFlowProvider';
+
+const ContextMenuContext = createContext(null);
+const { Provider } = ContextMenuContext;
+
+export const useContextMenu = () => useContext(ContextMenuContext);
+
+const ContextMenuProvider = ({ children, service }) => {
+  const { create, dismiss } = useDialog();
+  const { show: showLabellingFlow } = useLabellingFlow();
+
+  /**
+   * Sets the implementation of a context menu service that can be used by extensions.
+   *
+   * @returns void
+   */
+  useEffect(() => {
+    if (service) {
+      service.setServiceImplementation({
+        show,
+        hide,
+      });
+    }
+  }, [hide, service, show]);
+
+  const hide = useCallback(() => dismiss({ id: 'context-menu' }), [dismiss]);
+
+  /**
+   * Show the context menu and override its configuration props.
+   *
+   * @param {ContextMenuProps} props { eventData, isTouchEvent, onClose, visible }
+   * @returns void
+   */
+  const show = useCallback(
+    ({ event }) => {
+      hide();
+      create({
+        id: 'context-menu',
+        isDraggable: false,
+        useLastPosition: false,
+        content: ToolContextMenu,
+        contentProps: {
+          eventData: event,
+          onClose: () => dismiss({ id: 'context-menu' }),
+          onSetLabel: (eventData, measurementData) =>
+            showLabellingFlow({
+              event: eventData,
+              centralize: true,
+              props: {
+                measurementData,
+                skipAddLabelButton: true,
+                editLocation: true,
+              },
+            }),
+          onSetDescription: (eventData, measurementData) =>
+            showLabellingFlow({
+              event: eventData,
+              centralize: false,
+              defaultPosition: _getDefaultPosition(eventData),
+              props: {
+                measurementData,
+                editDescriptionOnDialog: true,
+              },
+            }),
+        },
+        defaultPosition: _getDefaultPosition(event),
+      });
+    },
+    [create, dismiss, hide, showLabellingFlow]
```

Comment:
We may need to do this gradually, but I would think `showLabelingFlow` would not exist directly in `ContextMenuProvider`. When creating the context menu, you would probably want to be able to set the options in the context menu, and the behavior when they're clicked.

Kicking off the labeling flow would be configuration/props?

Reply:
Yeah, they are independent and they are using each other using context like other ui components use modal. Kicking off labelling flow would be through props when calling it through its provider show method or directly at component declaration.